File: src/main/java/land/leets/domain/contributor/domain/Contributor.java
package land.leets.domain.contributor.domain;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.FetchType;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import land.leets.domain.portfolio.domain.Portfolio;
import lombok.AccessLevel;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Entity(name = "contributors")
@Builder
@Getter
@Setter
@NoArgsConstructor(access = AccessLevel.PROTECTED)
@AllArgsConstructor
public class Contributor {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @JsonIgnore
    private Long id;

    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "portfolioId")
    @JsonIgnore
    private Portfolio portfolio;

    @Column(nullable = false)
    private String name;

    @Column(nullable = false)
    private Position position;

    @Column
    private String githubUrl;

    @Column
    private String profileUrl;

    @Column
    private String profile;
}

File: src/main/kotlin/land/leets/domain/contributor/domain/Contributor.kt
package land.leets.domain.contributor.domain

import com.fasterxml.jackson.annotation.JsonIgnore
import jakarta.persistence.*
import land.leets.domain.portfolio.domain.Portfolio

@Entity(name = "contributors")
class Contributor(
    @Column(nullable = false)
    val name: String,

    @Column(nullable = false)
    val position: Position,

    @Column
    val githubUrl: String? = null,

    @Column
    val profileUrl: String? = null,

    @Column
    val profile: String? = null,

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @JsonIgnore
    val id: Long = 0L,

    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "portfolioId")
    @JsonIgnore
    var portfolio: Portfolio? = null
) {
    fun updatePortfolio(portfolio: Portfolio) {
        this.portfolio = portfolio
    }
}
